website/src/components/editor/Rule/index.js

```diff
@@ -109,3 +138,127 @@ export function Rule({variations, label, isDefaultRule}) {
+  const getIcon = useCallback(() => {
+    switch (variant) {
+      case 'group':
+        return 'fa-xmark';
+      case 'rule':
+        return 'fa-minus';
+    }
+  }, [variant]);
+
+  return (
+    <button className={styles.removeButton} onClick={handleOnClick}>
+      <span className="fa-stack fa-1x">
+        <i className={clsx('fa-solid fa-circle fa-stack-2x', styles.bg)}></i>
+        <i className={`fa-solid ${getIcon()} fa-stack-1x fa-inverse`}></i>
+      </span>
+    </button>
+  );
+}
+
+/**
+ * Parses a JSON object into a custom query language based on the nikunjy/rules library.
+ *
+ * @param {Object} json - The JSON object representing the query.
+ *   @property {string} combinator - The combinator used to combine multiple rules (e.g., "AND", "OR").
+ *   @property {Array} rules - An array of rule objects.
+ *     @property {string} field - The field to apply the rule on.
+ *     @property {string} operator - The operator to use for the comparison.
+ *     @property {string} [value] - The value to compare when the operator requires it.
+ *     @property {string} [combinator] - The combinator used to combine multiple sub-rules within this rule (e.g., "AND", "OR").
+ *     @property {Array} [rules] - An array of sub-rule objects.
+ *
+ * @returns {string} - The custom query string.
+ */
+function parseJsonToCustomQuery(json) {
+  /**
+   * Recursive helper function to process rules.
+   *
+   * @param {Object} rule - The rule object to process.
+   *   @property {string} field - The field to apply the rule on.
+   *   @property {string} operator - The operator to use for the comparison.
+   *   @property {string} [value] - The value to compare when the operator requires it.
+   *   @property {string} [combinator] - The combinator used to combine multiple sub-rules within this rule (e.g., "AND", "OR").
+   *   @property {Array} [rules] - An array of sub-rule objects.
+   *
+   * @returns {string} - The custom query string for the rule.
+   */
+  function processRule(rule) {
+    let query = '';
+
+    if (rule.field && rule.operator) {
+      query += `${rule.field} ${rule.operator}`;
+
+      if (rule.value) {
+        if (rule.operator == 'in') query += ` [${rule.value}]`;
+        else query += ` ${rule.value}`;
+      }
+    }
+
+    if (rule.rules && rule.rules.length > 0) {
+      const subRules = rule.rules.map(processRule).join(` ${rule.combinator} `);
+      query += ` (${subRules}) `;
+    }
+
+    return query.trim();
+  }
+
+  if (!json.combinator || !json.rules || !Array.isArray(json.rules)) {
+    throw new Error('Invalid JSON format for the query.');
+  }
+
+  return json.rules.map(processRule).join(` ${json.combinator} `);
+}
+
+const ruleOperators = [
+  {name: '==', label: 'Equals To'},
+  {name: '!=', label: 'Not Equals To'},
+  {name: '<', label: 'Less Than'},
+  {name: '>', label: 'Greater Than'},
+  {name: '<=', label: 'Less Than Equal To'},
+  {name: '>=', label: 'Greater Than Equal To'},
```

Comment:
I am more in favour of using the text notation everywhere.
```suggestion
  {name: 'eq', label: 'Equals To'},
  {name: 'ne', label: 'Not Equals To'},
  {name: 'lt, label: 'Less Than'},
  {name: 'gt', label: 'Greater Than'},
  {name: 'le', label: 'Less Than Equal To'},
  {name: 'ge', label: 'Greater Than Equal To'},
```